src/features/farming/town/components/GoblinVillageModal.tsx

```diff
@@ -13,6 +22,82 @@ export const GoblinVillageModal: React.FC<{ onClose: () => void }> = ({
     window.location.href = `${window.location.pathname}#/goblins`;
   };
 
+  const { gameService } = useContext(Context);
+  const [game] = useActor(gameService);
+  const [isLoading, setIsLoading] = useState(true);
+  const [inventory, setInventory] = useState<Inventory>({});
+  const [balance, setBalance] = useState<Decimal>(new Decimal(0));
+  const [showSyncCaptcha, setShowCaptcha] = useState(false);
+  const localInventory = JSON.stringify(
```

Comment:
I would suggest abstracting this into a function for readability.

e.g.

```function inventoryMatch(...){
   return ....
}```

Reply:
The abstraction is also on my radar, and part of what I want to combine with the comment about the Delta and specific items lost in the event of a mint/chain sync.